framework/src/media/MediaPlayerImpl.cpp

```diff
@@ -143,59 +141,63 @@ void MediaPlayerImpl::preparePlayer(player_result_t &ret)
 	medvdbg("MediaPlayer Worker : preparePlayer\n");
 	if (mCurState != PLAYER_STATE_IDLE) {
 		meddbg("MediaPlayer prepare fail : wrong state\n");
-		goto errout;
+		ret = PLAYER_ERROR_INVALID_STATE;
+		return notifySync();
 	}
 	if (mInputDataSource == nullptr) {
 		meddbg("MediaPlayer prepare fail : mInputDataSource is not set\n");
-		goto errout;
+		ret = PLAYER_ERROR_INVALID_PARAMETER;
+		return notifySync();
 	}
 
 	if (!mInputDataSource->open()) {
-		meddbg("MediaPlayer prepare fail : file open fail\n");
-		goto errout;
+		meddbg("MediaPlayer prepare fail : open fail\n");
+		ret = PLAYER_ERROR_FILE_OPEN_FAILED;
+		return notifySync();
 	}
 
 	if (set_audio_stream_out(mInputDataSource->getChannels(), mInputDataSource->getSampleRate(),
 							 mInputDataSource->getPcmFormat()) != AUDIO_MANAGER_SUCCESS) {
 		meddbg("MediaPlayer prepare fail : set_audio_stream_out fail\n");
-		goto errout;
+		ret = PLAYER_ERROR_INTERNAL_OPERATION_FAILED;
+		return notifySync();
 	}
 
 	mBufSize = get_output_frames_to_byte(get_output_frame_count());
 	if (mBufSize < 0) {
 		meddbg("MediaPlayer prepare fail : get_output_frames_byte_size fail\n");
-		goto errout;
+		ret = PLAYER_ERROR_INTERNAL_OPERATION_FAILED;
+		return notifySync();
 	}
 
 	medvdbg("MediaPlayer mBuffer size : %d\n", mBufSize);
 
 	mBuffer = new unsigned char[mBufSize];
 	if (!mBuffer) {
 		meddbg("MediaPlayer prepare fail : mBuffer allocation fail\n");
-		goto errout;
+		if (get_errno() == ENOMEM) {
+			ret = PLAYER_ERROR_OUT_OF_MEMORY;
+		} else {
+			ret = PLAYER_ERROR_INVALID_OPERATION;
```

Comment:
I Think internal fail is more good option.

Reply:
Fixed it